### PR TITLE
remove non-repeating jobs after completion, add tests and .eslintrc

### PR DESCRIPTION
**Pull request by bywo**

### CODE REVIEW

File: test/integration.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      .create(jobType, {})
+      .schedule('in 10 minutes')
+      .save(function(err) {
+        if (err) {
+          return done(err);
+        }
+
+        mongo.collection('agendaJobs').find().toArray(function(err, arr) {
+          if (err) {
+            throw err;
```
Comment (jetzhou) [inline test/integration.js]: `return done(err)` would be faster, right?

File: test/integration.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      .create(jobType, {})
+      .schedule('now')
+      .save(function(err) {
+        if (err) {
+          return done(err);
+        }
+
+        mongo.collection('agendaJobs').find().toArray(function(err, arr) {
+          if (err) {
+            throw err;
```
Comment (jetzhou) [inline test/integration.js]: same

File: test/integration.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          }
+          expect(arr).to.have.length(1);
+        });
+
+        setTimeout(function() {
+          mongo.collection('agendaJobs').find().toArray(function(err, arr) {
+            if (err) {
+              throw err;
```
Comment (jetzhou) [inline test/integration.js]: same

File: test/integration.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        mongo.collection('agendaJobs').find().toArray(function(err, arr) {
+          if (err) {
+            throw err;
+          }
+          expect(arr).to.have.length(1);
+        });
+
+        setTimeout(function() {
```
Comment (jetzhou) [inline test/integration.js]: `agenda`/`jobs` emits a complete event, I believe we can do a `jobs.on("complete", function)` here, right?

Comment (bywo) [inline test/integration.js]: ahh good catch. will do!

**Pull request merged**